src/pkt_checker.rs

```diff
@@ -0,0 +1,206 @@
+use std::{
+    collections::{HashMap, LinkedList},
+    sync::{atomic::{AtomicBool, Ordering}, mpsc::Sender, Arc, Mutex, RwLock},
+};
+
+use crate::{
+    op_ctx::ReadOpCtx,
+    recv_pkt_map::RecvPktMap,
+    responser::{RespAckCommand, RespCommand},
+    types::{Msn, Psn},
+    Error,
+};
+
+use log::{error, info};
+
+#[derive(Debug)]
+pub(crate) struct PacketChecker {
+    thread: Option<std::thread::JoinHandle<()>>,
+    stop_flag: Arc<AtomicBool>,
+}
+
+impl PacketChecker {
+    pub(crate) fn new(
+        send_queue: Sender<RespCommand>,
+        recv_pkt_map: Arc<RwLock<HashMap<Msn, Arc<Mutex<RecvPktMap>>>>>,
+        read_op_ctx_map: Arc<RwLock<HashMap<Msn, ReadOpCtx>>>,
+    ) -> Self {
+        let ctx = PacketCheckerContext {
+            send_queue,
+            recv_pkt_map,
+            read_op_ctx_map,
+        };
+        let stop_flag = Arc::new(AtomicBool::new(false));
+        let thread_stop_flag = Arc::clone(&stop_flag);
+        let thread = std::thread::spawn(move || {
+            PacketCheckerContext::working_thread(&ctx, &thread_stop_flag);
+        });
+        Self {
+            thread: Some(thread),
+            stop_flag,
+        }
+    }
+}
+
+impl Drop for PacketChecker {
+    fn drop(&mut self) {
+        self.stop_flag.store(true, Ordering::Relaxed);
+        if let Some(thread) =  self.thread.take(){
+            if let Err(e) = thread.join(){
+                error!("Failed to join the WorkDescPoller thread: {:?}", e);
+            }
+        }
+    }
+}
+
+struct PacketCheckerContext {
+    send_queue: Sender<RespCommand>,
+    recv_pkt_map: Arc<RwLock<HashMap<Msn, Arc<Mutex<RecvPktMap>>>>>,
+    read_op_ctx_map: Arc<RwLock<HashMap<Msn, ReadOpCtx>>>,
+}
+
+impl PacketCheckerContext {
+    fn working_thread(ctx: &Self, stop_flag: &AtomicBool) {
+        while !stop_flag.load(Ordering::Relaxed) {
+            if let Err(e) = ctx.check_pkt_map() {
+                error!("PacketChecker stopped: {:?}", e);
+                break;
+            }
+        }
```

Comment:
Here please also log that the worker is stopped by the `stop_flag`.

Reply:
I will add that.